File: easy/go/0020.go
func isValid(s string) bool {
    var stack []byte;
    var idx int = 0;
    for len(s) > 0{
        if s[0] == '('{
            stack = append(stack, ')')
            idx++
        }else if s[0] == '['{
            stack = append(stack, ']')
            idx++
        }else if s[0] == '{'{
            stack = append(stack, '}')
            idx++
        }else if idx == 0 || stack[idx-1] != s[0]{
            return false
        }else{
            stack = stack[:len(stack)-1]
            idx--
        }
        s = s[1:]
    }
    if idx != 0 {
        return false
    }
    return true
    
}

File: easy/cpp/0020.cpp
class Solution {
public:
    bool isValid(string s) {
        vector<char> stack;
        map<char, char>table;
        table['('] = ')';
        table['['] = ']';
        table['{'] = '}';
        int idx = 0;
        while(s[idx]){
            if(s[idx] == '(' || s[idx] == '[' || s[idx] == '{'){
                stack.push_back(table[s[idx]]);
            }else if(stack.size() == 0 ||stack.back() != s[idx]){
                return false;
            }else{
                stack.pop_back();
            }
            idx++;
        }
        if(stack.size())return false;
        return true;
    }
};